config/mfDefsGEM.mk

```diff
@@ -1,17 +1,25 @@
-include $(XDAQ_ROOT)/config/mfAutoconf.rules
-include $(XDAQ_ROOT)/config/mfDefs.$(XDAQ_OS)
+XDAQ_ROOT?=/opt/xdaq
+XHAL_ROOT?=/opt/xhal
+uHALROOT?=/opt/cactus
 
-BUILD_HOME := $(shell cd $(BUILD_HOME); pwd)
+ifndef BUILD_SUPPORT
+BUILD_SUPPORT=build
+endif
+
+include $(XDAQ_ROOT)/$(BUILD_SUPPORT)/mfAutoconf.rules
+include $(XDAQ_ROOT)/$(BUILD_SUPPORT)/mfDefs.$(XDAQ_OS)
+
+BUILD_HOME?= $(shell pwd)/..
```

Comment:
Why also defining `BUID_HOME` in the `.mk` files? Does this definition assume that the `.mk` is included at a given depth in the folder tree?

Reply:
this allows to call `make` from the subdirectory itself and get the correct behaviour

Reply:
Right, but doesn't it work only for a first level sub-directory? For example, does it have the correct behavior for `Makefile` in `gemhardware` and `gemhardware/devices`?

I just tried that example and it works... because `BUILD_HOME` is defined in the `Makefile` themselves:
``` bash
[lpetre@gemdaq-build-xdaq15 cmsgemos]$ grep 'BUILD_HOME?=' gemhardware/Makefile
BUILD_HOME?= $(shell dirname `cd ./; pwd`)
[lpetre@gemdaq-build-xdaq15 cmsgemos]$ grep 'BUILD_HOME?=' gemhardware/devices/Makefile
BUILD_HOME?= $(shell dirname `cd ../; pwd`)
```
Looks like to be an exception for the `gemhardware/*` sub-folders though. (I don't like that kind of exception, but, okay, this is clearly not a showstopper.)

Reply:
Right, but doesn't it work only for a first level sub-directory? For example, does it have the correct behavior for `Makefile` in `gemhardware` and `gemhardware/devices`?

I just tried that example and it works... because `BUILD_HOME` is defined in the `Makefile` themselves:
``` bash
[lpetre@gemdaq-build-xdaq15 cmsgemos]$ grep 'BUILD_HOME?=' gemhardware/Makefile
BUILD_HOME?= $(shell dirname `cd ./; pwd`)
[lpetre@gemdaq-build-xdaq15 cmsgemos]$ grep 'BUILD_HOME?=' gemhardware/devices/Makefile
BUILD_HOME?= $(shell dirname `cd ../; pwd`)
```
Looks like to be an exception for the `gemhardware/*` sub-folders though. (I don't like that kind of exception, but, okay, this is clearly not a showstopper.)

Reply:
``BUILD_HOME?=$(shell dirname `cd appropriate path; pwd`)`` is (should be) present in all package/subpackage `Makefile`s

Reply:
I naively `grep`'ed `BUILD_HOME?=` while in the other package `Makefile`s `BUILD_HOME:=` is used. But then, I'm not convinced why a definition of `BUILD_HOME` is required in the `config/*.mk` files. The variable is already defined in the "parent" `Makefile`.